### PR TITLE
rec: Delint pdns recursor.cc

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/pdns_recursor.cc [hunk 1]
```diff
@@ -900,37 +906,38 @@ static uint32_t capPacketCacheTTL(const struct dnsheader& hdr, uint32_t ttl, boo
   return ttl;
 }
 
-void startDoResolve(void* p) // NOLINT(readability-function-cognitive-complexity): https://github.com/PowerDNS/pdns/issues/12791
+void startDoResolve(void* arg) // NOLINT(readability-function-cognitive-complexity): https://github.com/PowerDNS/pdns/issues/12791
 {
-  auto dc = std::unique_ptr<DNSComboWriter>(reinterpret_cast<DNSComboWriter*>(p));
-  SyncRes sr(dc->d_now);
+  auto comboWriter = std::unique_ptr<DNSComboWriter>(reinterpret_cast<DNSComboWriter*>(arg)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
```
Comment (rgacogne) [inline pdns/recursordist/pdns_recursor.cc:911]:
```suggestion
  auto comboWriter = std::unique_ptr<DNSComboWriter>(static_cast<DNSComboWriter*>(arg));
```

**Review by rgacogne — APPROVED** [review]

File: pdns/recursordist/pdns_recursor.cc [hunk 2]
```diff
@@ -2140,7 +2158,7 @@ static string* doProcessUDPQuestion(const std::string& question, const ComboAddr
 #endif
 
     // We do not have a SyncRes specific Lua context at this point yet, so ok to use t_pdl
-    if (needECS || (t_pdl && (t_pdl->d_gettag || t_pdl->d_gettag_ffi)) || dh->opcode == Opcode::Notify) {
+    if (needECS || (t_pdl && (t_pdl->d_gettag || t_pdl->d_gettag_ffi)) || dnsheader->opcode == static_cast<unsigned>(Opcode::Notify)) {
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/recursordist/pdns_recursor.cc]: Better indeed, pushed.

**Review by rgacogne — APPROVED** [review]

**Comment by omoerbeek** [comment]
grumble, locally it was clean... but CI gives me a couple of:
```
build recursor (tsan): pdns/recursordist/pdns_recursor.cc#L1120
narrowing conversion from 'unsigned int' to signed type 'int' is implementation-defined (bugprone-narrowing-conversions - Level=Warning)
```

File: pdns/recursordist/pdns_recursor.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/recursordist/pdns_recursor.cc:2161]: And reverted, CI clang-tidy was not happy, though the local one is...

**Pull request merged**